test/js-api/memory/constructor.any.js

```diff
@@ -128,3 +131,33 @@ test(() => {
   const memory = new WebAssembly.Memory(argument, {});
   assert_Memory(memory, { "size": 0 });
 }, "Stray argument");
+
+test(() => {
+  const argument = { "initial": 1 };
+  const memory = new WebAssembly.Memory(argument);
+  assert_equals(memory.type().index, "u32");
+}, "Memory with default constructor");
+
+test(() => {
+  const argument = { "initial": 1, "index": "u32" };
+  const memory = new WebAssembly.Memory(argument);
+  assert_equals(memory.type().index, "u32");
+}, "Memory with u32 index constructor");
+
+test(() => {
+  const argument = { "initial": 1, "index": "u64" };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 1 });
+  assert_equals(memory.type().index, "u64");
```

Comment:
This seems to depend on the js-types proposal:   https://github.com/WebAssembly/js-types/blob/main/proposals/js-types/Overview.md .  Is that intentional?   I guess we should probably not depend on that as it is not complete yet.

Reply:
I don't think so. It's a WebAssembly.Memory method that returns "memory type" object:
```javascript
var memory = new WebAssembly.Memory({ initial: 1, index: "u64"})
memory.type()
```
{minimum: 1, shared: false, index: 'u64'}

Reply:
Well, you are right. It's a part of the proposal. Should i remove these tests then?

Reply:
Perhaps we could detect the presence of the `type()` method at runtime and leave a comment about how this depends on js-types proposal.

Also, I wonder if you could include the index check in `assert_Memory` so this could be a single line:

```
 assert_Memory(memory, { "size": 1, "index": "u64" });
```

Reply:
> I guess so yes, is there any other way to verify the type I wonder?

Compile a small wasm test with 64bit memory.init, embed it into JS as UInt8Array and try running with that memory. But it looks like an overkill. WebAssembly.Memory does not export any other info to JS

Reply:
There's also a slight problem with checking type() as i did not think it's part of another proposal and updated V8 code to include the "index" field. So to include that, js-types proposal should be updated

Reply:
I guess it will be part of whichever proposal lands first.